etcdmain/grpc_proxy.go

```diff
@@ -70,11 +73,28 @@ func newGRPCProxyStartCommand() *cobra.Command {
 	cmd.Flags().StringVar(&grpcProxyCert, "cert", "", "identify secure connections with etcd servers using this TLS certificate file")
 	cmd.Flags().StringVar(&grpcProxyKey, "key", "", "identify secure connections with etcd servers using this TLS key file")
 	cmd.Flags().StringVar(&grpcProxyCA, "cacert", "", "verify certificates of TLS-enabled secure etcd servers using this CA bundle")
+	cmd.Flags().BoolVar(&grpcProxySelfAddr, "self-addr", false, "true to replace members in member list response")
```

Comment:
@heyitsanthony So `self-addr` is not specified, `MemberList` just returns its own listen client urls, never returning actual cluster member lists?